setup.py

```diff
@@ -435,8 +433,14 @@ def run(self):
 else:
     extra_compile_args = ["-Werror"]
     extra_link_args = []
-    if debugging_symbols_requested:
-        extra_compile_args.append("-g")
+    if not debugging_symbols_requested:
+        # Strip debugging symbols (included by default)
+        extra_compile_args.append("-g0")
+    else:
+        # TODO: these should override the defaults provided by Python
```

Comment:
distutils adds NDEBUG and -O3 by default it seems without a feasible way to remove those compilation flags. Appending these at the end should override those according to the SO link shared above

Reply:
I don't recommend building with `-O0` actually as it hides a lot of problems. In case of `-O0` memory is mored often zeroed and thus invalid memory access is not that fatal (in the higher `-Ox` cases you would get random data). This makes the detection of bugs a lot harder. It might be better to keep the optimization level here and add flags that make debugging easier. Personally I like `-ggdb -fno-omit-frame-pointer`. This gives better stacktraces and a bit more debugging information.

Reply:
gdb suggests turning off optimizations:

https://sourceware.org/gdb/onlinedocs/gdb/Optimized-Code.html

There are certainly exceptions but I think as a general rule (especially for people that aren't super well versed in debugging the extensions yet) that no optimizations will be easier to follow

Reply:
Yes, the debug information with `-O0` is definitely better, I would like to leave `-fno-omit-frame-pointer` somewhere here in the document as this is the option that provides the most debug information for the smallest performance hit and is really useful if you have an memory-out-of-bounds issue that wouldn't occur easily in the unoptimized version.

Reply:
Sounds good. I think this is off by default with -O0 per the docs but doesn't hurt to add again

https://gcc.gnu.org/onlinedocs/gcc-3.4.4/gcc/Optimize-Options.html